view/js/linkPreview.js

```diff
@@ -512,7 +512,7 @@
 		var getAttachmentData = function(content) {
 			var data = {};
 
-			var match = content.match(/(.*)\[attachment(.*?)\](.*?)\[\/attachment\](.*)/ism);
+			var match = content.match(/(.*)\[attachment(.*?)\](.*?)\[\/attachment\](.*)/im);
```

Comment:
I've tested this regular expression with the input text:
```
Before attachment

[attachment]Attachment[/attachment]

After attachment
```
Result with `/(.*)\[attachment(.*?)\](.*?)\[\/attachment\](.*)/im`:
```
[
  "[attachment]Attachment[/attachment]",
  "",
  "",
  "Attachment",
  ""
]
```
Result with `/(.*)\[attachment(.*?)\](.*?)\[\/attachment\](.*)/ism`:
```
[
  "Before attachment

[attachment]Attachment[/attachment]

After attachment",
  "Before attachment

",
  "",
  "Attachment",
  "

After attachment"
]
```

It all depends if you're making any use of the before and after capturing groups.